src/corva/models/base.py

```diff
@@ -5,64 +5,62 @@
 
 import pydantic.generics
 
+from corva import utils
 from corva.api import Api
 from corva.configuration import Settings
-from corva.state.redis_adapter import RedisAdapter
 from corva.state.redis_state import RedisState
 
 
-class CorvaModelConfig:
-    arbitrary_types_allowed = True
-    extra = pydantic.Extra.allow
-    validate_assignment = True
+class EventConfig:
+    extra = pydantic.Extra.ignore
+    allow_mutation = False
 
 
-class CorvaBaseModel(pydantic.BaseModel):
-    Config = CorvaModelConfig
+class CorvaBaseEvent(pydantic.BaseModel):
+    class Config(EventConfig):
+        pass
+
+
+class CorvaBaseGenericEvent(pydantic.generics.GenericModel):
+    class Config(EventConfig):
+        pass
 
 
-class BaseEvent(CorvaBaseModel, ABC):
+class RawBaseEvent(ABC):
     @staticmethod
     @abstractmethod
-    def from_raw_event(event: Any) -> Union[List[BaseEvent], BaseEvent]:
+    def from_raw_event(event: Any) -> Union[List[RawBaseEvent], RawBaseEvent]:
         pass
 
 
-BaseEventTV = TypeVar('BaseEventTV', bound=BaseEvent)
+RawBaseEventTV = TypeVar('RawBaseEventTV', bound=RawBaseEvent)
 
 
-class BaseContext(pydantic.generics.GenericModel, Generic[BaseEventTV]):
+class BaseContext(pydantic.generics.GenericModel, Generic[RawBaseEventTV]):
     """Stores common data for running a Corva app."""
 
-    Config = CorvaModelConfig
+    class Config:
+        arbitrary_types_allowed = True
+        extra = pydantic.Extra.ignore
+        validate_assignment = True
 
-    event: BaseEventTV
+    event: RawBaseEventTV
     settings: Settings
     api: Api
-    _cache: Optional[RedisState] = None
-
-    user_result: Any = None
-
+    cache_: Optional[RedisState] = None
```

Comment:
Why was this renamed to an underscore with a suffix?

Reply:
Pydantic does some magic with fields prefixed with `_` like `_cache`. For some reason tests started to fail for old name.